Workiva/ixbrl-viewer
# XT-3013: Create ixbrl-viewer review mode

#### Reason for change
Closes #453 

#### Description of change

- Added `reviewMode` to `iXBRLViewer.options`
  - Populated from query parameter
  - Can be used to conditionally perform review-mode specific tasks
- ~Added text to banner when review mode is enabled~
- When review mode is enabled:
  - Preprocessing for untagged elements occurs
  - Additional options are added to the toolbar
  - Highlighting key changes

Note: `number-matcher*` files were brought over from Workiva viewer and changed as little as possible to conform to `ixbrl-viewer` environment without making functional changes. 

![Screenshot 2023-07-18 at 12 48 23 PM](https://github.com/Workiva/ixbrl-viewer/assets/105066394/c1332b79-0726-4c10-be54-018fd6f82705)

#### Steps to Test

- Generate three viewers:
  - A: One using `master` to demonstrate current default behavior
  - B: One using Workiva's viewer to demonstrate intended review mode behavior
  - C: One using this branch
 - Compare A to C without review mode, confirm highlighting looks the same
 - Compare B to C with review mode enabled, confirm highlighting looks the same

**review**:
@Workiva/xt
@paulwarren-wk
